pkg/controller/installation/core_controller.go

```diff
@@ -914,6 +918,13 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
+	k8sEndpoint, err := utils.GetK8sServiceEndPoint(r.client)
```

Comment:
Maybe I am missing context, but what is responsible for creating this endpoint? This is the first controller that reconciles networking components for calico/tigera, so if the endpoint is not present, networking fails to be installed. This will cascade and everything fails to be installed. Does this mean it will break installations in new clusters?